src/utils/session-core.js

```diff
@@ -181,9 +189,13 @@ export const uploadDocument = async (session, uploadType, fileObject, otherPodUs
   // Place file into Pod container and generate new ttl file for container
   await placeFileInContainer(session, fileObject, containerUrl);
   const newTtlFile = await createResourceTtlFile(fileObject, documentUrl);
+  const signingKey = verifyDocument ? await getUserSigningKey(session) : null;
```

Comment:
Have tested out the verify file feature. It seems to work out nicely as noted from what's stated from the PR with the private/public keys and the signature.ttl.

Was wonder if we should relegate the initial generation of the keys to when people log in for the first time like what we're doing with the Users container in App.jsx.

I've tried uploading for the first time without PASS Credentials and it was quite slow since it had to generate the keys first before creating a signature for the file for uploading. It eventually finished, but I though the process hanged.

Subsequent uploads after having the keys were much faster. So that's all good.

Reply:
I haven't thought too much about the UX for this feature yet. I'm not sure how I feel about doing this on login though. I think the biggest challenge with the signing UX is distinguishing between users who need credentials (case managers who sign things) and users who don't (clients who are uploading their own documents to their own pods). One of the drawbacks of this design is it adds PASS-specific files to a user's pod, and I'd like to avoid adding those files if the user doesn't need them.

I think this is difficult to do on login because login doesn't give us a way to identify what type of user this person is. It just lets us access their pod. We'd need to do a separate fetch to find a document that tells us that.

We could have a separate button for generating credentials on a user profile management page. Then if they attempt to verify a document without credentials, we show a popup that says "Your profile has no associated credentials. Please generate them before signing" and stop verifying the document. That would make it clearer to the user where the slowdowns are occurring.

Reply:
Considering that it only occurs once for when the user first uploads without their keys, I think this would do for now.

We could revisit this again in the future. I'll leave that discretion to you.